craft_application/models/project.py

```diff
@@ -115,6 +122,36 @@ def effective_base(self) -> Any:  # noqa: ANN401 app specific classes can improv
             return self.base
         raise RuntimeError("Could not determine effective base")
 
+    @classmethod
+    @abc.abstractmethod
+    def _providers_base(
+        cls, base: str | None
+    ) -> craft_providers.bases.BaseAlias | None:
+        """Get a BaseAlias from the application's base.
+
+        :param base: The application-specific base name.
+
+        :returns: The BaseAlias for the base or None if not found.
```

Comment:
At first glance I'd rather raise an exception than returning `None`, but I'm open to being convinced otherwise. What's the use case where not finding a `BaseAlias` wouldn't be an error? (And is it common enough that we want to have to check for None every time we use this vs. checking for an exception when not having a base alias is valid?)

Reply:
True. If we can't find the BaseAlias then it should be an error.
This would make the code safer and more readable.  I'll give it a try.